### ISSUE TITLE
Exception caught when trying to sync all data: unknown extension ?<p at position 2

### ISSUE THREAD

**Issue by alexifrim** [splitio/python-client#281]
Hello.

Tested on: 9.1.3 and 9.2.0 (multi-threaded mode)
On a specific splitio environment, the python sync fails with the following error
<img width="876" alt="image" src="https://user-images.githubusercontent.com/420564/200575336-260e923a-801d-43d7-bda1-3fb52e29a5d2.png">

This happens only for the python library (kotlin and others work), and as far as I can tell, it fails when syncing the segments.

Not sure if related, but on a different environment where the above error doesn't show, there's instead a different issue related to segments, `Tried to retrieve nonexistant segment _segment_name. Skipping`. This happens for a big list segments that exist.

I'll try to add more details if I can find any.

**Comment by mredolatti** [comment]
Hi @alexifrim, sorry to hear you're having issues with our SDK. Could you please set the logging level to debug and send us the output when that happens? that should provide us with a stack trace that should help us figure out what's going on.

Thank you very much!
Martin

**Comment by alexifrim** [comment]
Thanks for the quick reply. This is the trace:
```2022-11-08T17:45:20,436 DEBUG [MainProcess] [87064] Error:  
Traceback (most recent call last):
  File "/Users/alexandruifrim/.local/share/virtualenvs/rooms-configs-VUweeRAL/lib/python3.9/site-packages/splitio/sync/synchronizer.py", line 292, in sync_all
    if not self.synchronize_splits(None, False):
  File "/Users/alexandruifrim/.local/share/virtualenvs/rooms-configs-VUweeRAL/lib/python3.9/site-packages/splitio/sync/synchronizer.py", line 270, in synchronize_splits
    for segment in self._split_synchronizers.split_sync.synchronize_splits(till):
  File "/Users/alexandruifrim/.local/share/virtualenvs/rooms-configs-VUweeRAL/lib/python3.9/site-packages/splitio/sync/split.py", line 121, in synchronize_splits
    successful_sync, remaining_attempts, change_number, segment_list = self._attempt_split_sync(fetch_options,
  File "/Users/alexandruifrim/.local/share/virtualenvs/rooms-configs-VUweeRAL/lib/python3.9/site-packages/splitio/sync/split.py", line 103, in _attempt_split_sync
    change_number, segment_list = self._fetch_until(fetch_options, till)
  File "/Users/alexandruifrim/.local/share/virtualenvs/rooms-configs-VUweeRAL/lib/python3.9/site-packages/splitio/sync/split.py", line 76, in _fetch_until
    parsed = splits.from_raw(split)
  File "/Users/alexandruifrim/.local/share/virtualenvs/rooms-configs-VUweeRAL/lib/python3.9/site-packages/splitio/models/splits.py", line 237, in from_raw
    [condition.from_raw(c) for c in raw_split['conditions']],
  File "/Users/alexandruifrim/.local/share/virtualenvs/rooms-configs-VUweeRAL/lib/python3.9/site-packages/splitio/models/splits.py", line 237, in <listcomp>
    [condition.from_raw(c) for c in raw_split['conditions']],
  File "/Users/alexandruifrim/.local/share/virtualenvs/rooms-configs-VUweeRAL/lib/python3.9/site-packages/splitio/models/grammar/condition.py", line 126, in from_raw
    matcher_objects = [matchers.from_raw(x) for x in raw_condition['matcherGroup']['matchers']]
  File "/Users/alexandruifrim/.local/share/virtualenvs/rooms-configs-VUweeRAL/lib/python3.9/site-packages/splitio/models/grammar/condition.py", line 126, in <listcomp>
    matcher_objects = [matchers.from_raw(x) for x in raw_condition['matcherGroup']['matchers']]
  File "/Users/alexandruifrim/.local/share/virtualenvs/rooms-configs-VUweeRAL/lib/python3.9/site-packages/splitio/models/grammar/matchers/__init__.py", line 67, in from_raw
    return builder(raw_matcher)
  File "/Users/alexandruifrim/.local/share/virtualenvs/rooms-configs-VUweeRAL/lib/python3.9/site-packages/splitio/models/grammar/matchers/base.py", line 27, in __init__
    self._build(raw_matcher)
  File "/Users/alexandruifrim/.local/share/virtualenvs/rooms-configs-VUweeRAL/lib/python3.9/site-packages/splitio/models/grammar/matchers/string.py", line 240, in _build
    self._regex = re.compile(self._data)
  File "/Users/alexandruifrim/.pyenv/versions/3.9.13/lib/python3.9/re.py", line 252, in compile
    return _compile(pattern, flags)
  File "/Users/alexandruifrim/.pyenv/versions/3.9.13/lib/python3.9/re.py", line 304, in _compile
    p = sre_compile.compile(pattern, flags)
  File "/Users/alexandruifrim/.pyenv/versions/3.9.13/lib/python3.9/sre_compile.py", line 788, in compile
    p = sre_parse.parse(p, flags)
  File "/Users/alexandruifrim/.pyenv/versions/3.9.13/lib/python3.9/sre_parse.py", line 955, in parse
    p = _parse_sub(source, state, flags & SRE_FLAG_VERBOSE, 0)
  File "/Users/alexandruifrim/.pyenv/versions/3.9.13/lib/python3.9/sre_parse.py", line 444, in _parse_sub
    itemsappend(_parse(source, state, verbose, nested + 1,
  File "/Users/alexandruifrim/.pyenv/versions/3.9.13/lib/python3.9/sre_parse.py", line 749, in _parse
    raise source.error("unknown extension ?<" + char,
re.error: unknown extension ?<p at position 2```

**Comment by agustinona** [comment]
Hi @alexifrim ,

I believe I have tracked this issue down to a regular expression in use in one of your split definitions. To avoid sharing any details about your Split implementation here, I'd like to move this into a support ticket. You should have received an email from support@split.io. Please let me know if you didn't.

**Comment by alexifrim** [comment]
I received the email. Thank you!

**Comment by chillaq** [comment]
Hi @alexifrim , this issue is resolved in version 9.2.1, we removed the dummy segment check.

Please let us know if you have any questions.
